### PR TITLE
Added support converting unmarshalled configs to `interface{}`

### PR DESCRIPTION
**Pull request by BoskyWSMFN**
Fixed problem described in #459.
Nothing is broken so far.

### CODE REVIEW

File: internal/convert/converter.go [hunk 1]
```diff
@@ -91,6 +91,10 @@ func (c Converter) convert(name string, from any, toVal reflect.Value) error { /
 		return c.convertString(name, fromVal, toVal)
 	case toVal.Kind() == reflect.Struct:
 		return c.convertStruct(name, fromVal, toVal)
+	case toVal.Kind() == reflect.Interface: // Right after all other checks.
+		toVal.Set(fromVal)
```
[the remark this answers is not in the document]
Comment (BoskyWSMFN) [reply internal/convert/converter.go:95]: Didn't get your point. Special handling is required for maps, asaik, but it is the caller's responsibility to deal with keys and values if the destination contains fields of type interface{}.

[the remark this answers is not in the document]
Comment (ktong) [reply internal/convert/converter.go:95]: Sorry for confusion. To support case sensitive map key, the key in internal value has been wrapped with a special struct (https://github.com/nil-go/konf/blob/main/config.go#L166). So when unmarshaling back to map, it needs to unwrap the key back to string (case sensitive or not depends on configuration).

File: internal/convert/converter_test.go [hunk 2]
```diff
@@ -896,13 +902,85 @@ func TestConverter(t *testing.T) { //nolint:maintidx
...
+			expected:    pointer(any("str")),
+		},
+		{
+			description: "float to interface",
+			from:        42.42,
+			to:          pointer(any(nil)),
+			expected:    pointer(any(42.42)),
 		},
+		{
+			description: "map to interface",
```
Comment (BoskyWSMFN) [inline internal/convert/converter_test.go:924]: Added few more tests to check if maps.Unpack call is required. Seems to be excessive.

Comment (ktong) [reply internal/convert/converter_test.go:924]: I may miss test for Unpack in convert package. Here is the code calls maps.Unpack https://github.com/nil-go/konf/blob/857c68e060f9e24b961d1abaff28175c91969461/internal/convert/converter.go#L355

Comment (ktong) [reply internal/convert/converter_test.go:924]: Just tested it from most outer https://github.com/nil-go/konf/blob/main/config_test.go#L102

File: internal/convert/converter.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (BoskyWSMFN) [reply internal/convert/converter.go]: Oh, there's no confusion. I just had doubts about whether special handling was really necessary here. I found a test case that clearly shows it is. =)

File: internal/convert/converter.go [hunk 3]
```diff
@@ -550,6 +552,55 @@ func (c Converter) convertStruct(name string, fromVal, toVal reflect.Value) erro
 	}
 }
 
+func (c Converter) convertInterface(name string, fromVal, toVal reflect.Value) error {
```
Comment (BoskyWSMFN) [inline internal/convert/converter.go:555]: Copied most of the code from `convertMap` method and slightly changed it.

File: internal/convert/converter.go [hunk 4]
```diff
@@ -554,6 +552,55 @@ func (c Converter) convertStruct(name string, fromVal, toVal reflect.Value) erro
...
+	switch fromVal.Kind() {
+	case reflect.Map:
```
Comment (ktong) [inline internal/convert/converter.go:557]: is it possible just creating a Map[string]any and then return the result from convert with new created map?

File: internal/convert/converter_test.go [hunk 5]
```diff
@@ -896,13 +902,100 @@ func TestConverter(t *testing.T) { //nolint:maintidx
...
+			expected: pointer(struct {
+				Key1 interface{}
+				Key2 interface{}
+			}{
+				Key1: "value1",
+				Key2: "value2",
+			}),
+		},
+		{
+			description: "packed KV and field to interface{}",
```
Comment (BoskyWSMFN) [inline internal/convert/converter_test.go:978]: This test case shows that special handling is necessary. Now, all complex data types will be treated as map[string]interface{}.

File: internal/convert/converter.go (hunk 4 above)
Comment (BoskyWSMFN) [reply internal/convert/converter.go:557]: Oh... Right, it is. Haste makes waste.

File: internal/convert/converter_test.go (hunk 2 above)
Comment (BoskyWSMFN) [reply internal/convert/converter_test.go:924]: > I may miss test for Unpack in convert package. Here is the code calls maps.Unpack
> 
> https://github.com/nil-go/konf/blob/857c68e060f9e24b961d1abaff28175c91969461/internal/convert/converter.go#L355

Seems to be tested indirectly.

**Review by ktong — APPROVED** [review]
LGTM. Thanks for the fixing.

**Comment by ktong** [comment]
@BoskyWSMFN may I merge it now?

**Comment by BoskyWSMFN** [comment]
> @BoskyWSMFN may I merge it now?

Sure.

**Pull request merged**